### PR TITLE
Adding a service to expose Operator SDK metrics

### PR DESCRIPTION
**Pull request by DharmitD**
## Description

Adding a service to expose Operator SDK metrics port

## How Has This Been Tested?

On a test cluster:

- Added this service config
- Port forwarded to the localhost port via the service:
`oc port-forward svc/data-science-pipelines-operator 8080:8080 -n data-science-pipelines-operator`
- Curled to the localhost to retrieve a list of Operator SDK metrics:
`curl -v http://localhost:8080/metrics`

Find a list of metrics returned by the curl command [here.](http://pastebin.test.redhat.com/1092224)

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by gregsheremeta** [comment]
content looks good to me. The only question I have is whether we'd separate out the service into its own config file. The other config directories seem to have resources be mostly separate (for example: https://github.com/opendatahub-io/data-science-pipelines-operator/tree/main/config/rbac), and I'm not familiar enough with the conventions. Perhaps another reviewer can make a suggestion.

File: config/manager/manager-service.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: data-science-pipelines-operator-controller-manager
```
Comment (HumairAK) [inline config/manager/manager-service.yaml:4]: can you shorten the name to `data-science-pipelines-operator`

**Review by HumairAK — CHANGES_REQUESTED** [review]

**Comment by HumairAK** [comment]
Can you confirm that the servicemonitor works? Confirm this by checking if the metrics show up in ocp monitoring stack. My guess is it won't work because it's missing the labels for the selector here: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/prometheus/monitor.yaml 

The `port` will also need to be updated to match the exposed port in the `service`.

**Comment by gregsheremeta** [comment]
> My guess is it won't work because it's missing the labels for the selector here

By "it", did you mean the service? The service has to have the same labels as the servicemonitor?

**Comment by HumairAK** [comment]
> By "it", did you mean the service? The service has to have the same labels as the servicemonitor? 

yeah the service monitor will select on the labels for the service

**Comment by HumairAK** [comment]
@DharmitD you should also test by port-forwarding the service, not the pod `oc port-forward service/$service_name` since that is the change you are making

File: config/manager/manager-service.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+spec:
+  ports:
+    - name: http
```
[the remark this answers is not in the document]
Comment (DharmitD) [reply config/manager/manager-service.yaml]: Done, updated the port name.

File: config/manager/manager-service.yaml (hunk 1 above)
Comment (DharmitD) [reply config/manager/manager-service.yaml]: Done, updated the service name.

**Comment by DharmitD** [comment]
> @DharmitD you should also test by port-forwarding the service, not the pod `oc port-forward service/$service_name` since that is the change you are making

I updated the service to add the same label as the service monitor. Tried port forwarding through the service:
`oc port-forward svc/data-science-pipelines-operator 8080:8080 -n data-science-pipelines-operator`

The port forward worked, further queried metrics using `curl`:
`curl http://localhost:8080/metrics`
Curl returned the same set of metrics.

**Comment by DharmitD** [comment]
> Can you confirm that the servicemonitor works? Confirm this by checking if the metrics show up in ocp monitoring stack. My guess is it won't work because it's missing the labels for the selector here: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/config/prometheus/monitor.yaml
> 
> The `port` will also need to be updated to match the exposed port in the `service`.

Added a ServiceMonitor config in https://github.com/opendatahub-io/data-science-pipelines-operator/pull/24 with labels matching to the above service config.
The ServiceMonitor works and metrics can be queried.

**Pull request merged**